tests/test_heterobatch.py

```diff
@@ -0,0 +1,441 @@
+    ],
+    indirect=['nifty_backend'],
+)
+def test_normalization(data, nifty_backend, nterms, Nf=1000):
+    """Check that the normalization modes work as expected"""
+
+    t_list = data['t']
+    y_list = data['y']
+    dy_list = data['dy']
+    fmin_list = data['fmin']
+    Nf_list = [Nf] * len(t_list)
+
+    for norm in ['standard', 'model', 'log', 'psd']:
+        # heterobatch
+        heterobatch_results = nifty_backend[1](
+            t_list=t_list,
+            y_list=y_list,
+            dy_list=dy_list,
+            fmin_list=fmin_list,
+            Nf_list=Nf_list,
+            normalization=norm,
+            nterms=nterms,
+        )
+
+        # Single series
+        standard_result_powers = []
+        for i in range(len(t_list)):
+            standard_result = nifty_ls.lombscargle(
+                t=t_list[i],
+                y=y_list[i],
+                dy=dy_list[i],
+                fmin=fmin_list[i],
+                Nf=Nf,
+                normalization=norm,
+                nterms=nterms,
+                backend='finufft'
+                if nifty_backend[0] == 'finufft_heterobatch'
+                else 'finufft_chi2',
+            )
+            standard_result_powers.append(standard_result.power)
+
+        np.testing.assert_allclose(heterobatch_results.powers, standard_result_powers)
+
+
+@pytest.mark.parametrize(
+    'data', [{'N_series': 100, 'N': 50, 'N_batch': 3}], indirect=['data']
+)
+@pytest.mark.parametrize('center_data', [True, False])
+@pytest.mark.parametrize(
+    'nifty_backend,nterms',
+    [
+        ('finufft_heterobatch', 1),
+        ('finufft_chi2_heterobatch', 2),
+    ],
+    indirect=['nifty_backend'],
+)
+def test_center_data(data, center_data, nifty_backend, nterms, Nf=1000):
+    """Check that the center_data parameter work as expected"""
+
+    t_list = data['t']
+    y_list = data['y']
+    dy_list = data['dy']
+    fmin_list = data['fmin']
+    Nf_list = [Nf] * len(t_list)
+
+    # heterobatch
+    heterobatch_results = nifty_backend[1](
+        t_list=t_list,
+        y_list=y_list,
+        dy_list=dy_list,
+        fmin_list=fmin_list,
+        Nf_list=Nf_list,
+        center_data=center_data,
+        nterms=nterms,
+    )
+
+    # Single series
+    standard_result_powers = []
+    for i in range(len(t_list)):
+        standard_result = nifty_ls.lombscargle(
+            t=t_list[i],
+            y=y_list[i],
+            dy=dy_list[i],
+            fmin=fmin_list[i],
+            Nf=Nf,
+            center_data=center_data,
+            nterms=nterms,
+            backend='finufft'
+            if nifty_backend[0] == 'finufft_heterobatch'
+            else 'finufft_chi2',
+        )
+        standard_result_powers.append(standard_result.power)
+
+    np.testing.assert_allclose(heterobatch_results.powers, standard_result_powers)
+
+
+@pytest.mark.parametrize(
+    'data', [{'N_series': 100, 'N': 50, 'N_batch': 3}], indirect=['data']
+)
```

Comment:
It looks like you set default values in the `data` fixture definition above; were you able to get that working? If so, we might not need these lines here.